src/components/data-table.jsx

```diff
@@ -178,11 +176,16 @@ const DataTable = ({
 DataTable.propTypes = {
   ...DataTableHead.propTypes,
   ...DataTableBody.propTypes,
+  /**
+   * Whether the table is compact
+   */
+  compact: PropTypes.bool,
```

Comment:
I believe you mentioned you might, eventually, want more levels of "compactness". Might it be more future-proof to have a `density` prop with, at the moment, only 'normal' (default) and 'compact' as accepted values to be able to add more in the future?
They could be exported as:
```js
export const COMPACT = Symbol('COMPACT');
export const NORMAL = Symbol('NORMAL');
```
(see [this article](https://spicefactory.co/blog/2019/03/26/how-to-avoid-the-boolean-trap-when-designing-react-components/))